rs/bitcoin/kyt/tests/tests.rs

```diff
@@ -17,26 +51,71 @@ fn kyt_wasm() -> Vec<u8> {
     )
 }
 
-fn setup_env() -> (Principal, PocketIc) {
-    let env = PocketIc::new();
+impl Setup {
+    fn new() -> Setup {
+        let controller = PrincipalId::new_user_test_id(1).0;
+        let env = PocketIc::new();
+
+        let caller = env.create_canister_with_settings(Some(controller), None);
+        env.add_cycles(caller, 100_000_000_000_000);
+        env.install_canister(
+            caller,
+            UNIVERSAL_CANISTER_WASM.to_vec(),
+            vec![],
+            Some(controller),
+        );
+
+        let kyt_canister = env.create_canister();
+        env.add_cycles(kyt_canister, 100_000_000_000_000);
+        env.install_canister(kyt_canister, kyt_wasm(), vec![], None);
+
+        Setup {
+            controller,
+            caller,
+            kyt_canister,
+            env,
+        }
+    }
+
+    fn submit_kyt_call(
+        &self,
+        method: &str,
+        args: Vec<u8>,
+        cycles: u128,
+    ) -> Result<RawMessageId, UserError> {
+        let payload = wasm()
+            .call_with_cycles(
+                PrincipalId(self.kyt_canister),
+                method,
+                call_args()
+                    .other_side(args)
+                    .on_reject(wasm().reject_message().reject()),
+                Cycles::new(cycles),
+            )
+            .build();
+        self.env
+            .submit_call(self.caller, self.controller, "update", payload)
+    }
+}
 
-    let kyt = env.create_canister();
-    env.add_cycles(kyt, 100_000_000_000_000);
-    env.install_canister(kyt, kyt_wasm(), vec![], None);
-    (kyt, env)
+fn decode<'a, T: CandidType + Deserialize<'a>>(result: &'a WasmResult) -> T {
+    match result {
+        WasmResult::Reply(bytes) => decode_one(bytes).unwrap(),
+        WasmResult::Reject(msg) => panic!("unexpected reject: {}", msg),
+    }
 }
 
 #[test]
 fn test_check_address() {
```

Comment:
nit: Maybe it would make sense to add a test ensuring that response clean-up (`transform`) works as expected? It's not clear to me what's the best way to do it: querying directly probably doesn't make much sense or we could test that adding headers to the mocked response doesn't change the outcome.

Reply:
Good idea! I added different headers to `additional_responses`, which should exercise the `transform` function.

Reply:
ha `additional_responses` seems like a nice feature of Pocket IC! Thanks for the new test coverage!